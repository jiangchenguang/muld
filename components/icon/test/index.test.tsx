import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import Icon from '..';

const EventDemo = (props: any): React.ReactElement => {
    return <Icon {...props} />;
};

describe('Icon', () => {
    afterEach(cleanup);

    it('prop name url', () => {
        const { asFragment } = render(
            <Icon name="https://avatar.chengfayun.com.cn/chengfayun-avatar/6ce849b003c18744b54d1a9913db75e9.png" />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('prop color', () => {
        const { container } = render(<EventDemo name="chat-o" color="#07c160" />);
        const button = container.querySelector('.mul-icon-chat-o');
        expect(button).toHaveStyle({
            color: '#07c160',
        });
    });

    it('prop dot', () => {
        const { asFragment } = render(<EventDemo name="chat-o" dot />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('prop badge', () => {
        const { asFragment } = render(<EventDemo name="chat-o" badge="99" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('prop size', () => {
        const { asFragment } = render(<EventDemo name="chat-o" size="30" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
