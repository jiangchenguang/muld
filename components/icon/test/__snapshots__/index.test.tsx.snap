// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon prop badge 1`] = `
<DocumentFragment>
  <i
    class="sc-fznZeY hEcVDw mul-icon mul-icon-chat-o"
    content="F02D"
    name="chat-o"
    style="font-size: 20px;"
  >
    <div
      class="sc-AxiKw dIpnrG mul-info"
    >
      99
    </div>
  </i>
</DocumentFragment>
`;

exports[`Icon prop dot 1`] = `
<DocumentFragment>
  <i
    class="sc-fznZeY hEcVDw mul-icon mul-icon-chat-o"
    content="F02D"
    name="chat-o"
    style="font-size: 20px;"
  >
    <div
      class="sc-AxiKw dIpnrG mul-info mul-info--dot"
    />
  </i>
</DocumentFragment>
`;

exports[`Icon prop name url 1`] = `
<DocumentFragment>
  <i
    class="sc-fznZeY dvqeWw mul-icon"
    name="https://avatar.chengfayun.com.cn/chengfayun-avatar/6ce849b003c18744b54d1a9913db75e9.png"
    style="font-size: 20px;"
  >
    <img
      class="mul-icon__image"
      src="https://avatar.chengfayun.com.cn/chengfayun-avatar/6ce849b003c18744b54d1a9913db75e9.png"
    />
  </i>
</DocumentFragment>
`;

exports[`Icon prop size 1`] = `
<DocumentFragment>
  <i
    class="sc-fznZeY hEcVDw mul-icon mul-icon-chat-o"
    content="F02D"
    name="chat-o"
    style="font-size: 30px;"
  />
</DocumentFragment>
`;
